File: src/main/java/com/github/soniex2/notebetter/config/util/JsonHelper.java
package com.github.soniex2.notebetter.config.util;

import com.google.gson.JsonArray;
import com.google.gson.JsonObject;
import net.minecraft.util.JsonUtils;

/**
 * @author soniex2
 */
public class JsonHelper {
    public static JsonArray getJsonArrayOrNull(JsonObject jsonObject, String key) {
        if (jsonObject.has(key)) {
            if (jsonObject.isJsonNull()) {
                return null;
            }
            return JsonUtils.getJsonArray(jsonObject.get(key), key);
        } else {
            return null;
        }
    }
}

File: src/main/scala/com/github/soniex2/notebetter/config/util/JsonHelper.scala
package com.github.soniex2.notebetter.config.util

import com.github.soniex2.notebetter.util.GsonScalaHelper.{WJsonArray, WJsonObject}

/**
  * @author soniex2
  */
object JsonHelper {
  @inline def optJsonArray(jsonObject: WJsonObject, key: String): Option[WJsonArray] = {
    jsonObject get key flatMap (_.asJsonArray)
  }
}